src/main.cpp

```diff
@@ -1405,6 +1569,9 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
     }
 
     // Check for duplicate inputs
+    //
+    // A transaction with one or more transparent inputs from coinbase transactions
+    // MUST have no transparent outputs (i.e. tx_out_count MUST be 0)
```

Comment:
This is not where this check happens (we only know the types of the inputs in `ContextualCheckInputs`, which is by definition contextual).
```suggestion
```